app/graphql/resolvers/offer_resolver.rb

```diff
@@ -9,15 +9,37 @@ def valid?
   end
 
   def run
-    partner.offers.find(@arguments[:id])
+    if partner? || admin? && partner.present?
```

Comment:
This basically says-- if the requesting JWT includes a `partner` (or `admin`) role, and you've passed in a `gravity_partner_id` argument, _then_ return just the partner's offer that matches the request. 

Otherwise we assume you're a user or admin looking for a single offer.

This means that the `gravity_partner_id` input is responsible for determining this scope.